### DISCUSSION TITLE
Error with cors in production

### DISCUSSION THREAD

**Discussion by RaimonxDev** [strapi/strapi#6688] [Legacy Topics]
Hello I have an application created in angular and my api with strapi.

my angular app works on domain.com and my strapi works on api.domain.com 

when I made the first request it generates this error: Access to XMLHttpRequest at **'https://api.domain.com/graphql' from origin 'https://domain.com' has been blocked by CORS policy: No 'Access-Control -Allow-Origin 'header is present on the requested resource.**

**but.....** 

when I make the second request or reload the browser works perfect
and it doesn't show me any cors error? What would be the problem here?

I work with Docker.

Strapi version v 3.0.2

also create the middleware.js file 

```js module.exports = {
  timeout: 100,
  load: {
    before: ["responseTime", "logger", "cors", "responses", "gzip"],
    order: [
      "Define the middlewares' load order by putting their name in this array is the right order"
    ],
    after: ["parser", "router"]
  },
  settings: {
   cors: {
      origin: '*',
    },
   public: {
      path: "./public",
      maxAge: 60000
    }
  }
};
`

**Comment by sanojsilva** [comment]
Try adding `enabled: true` to your cors settings object.

**Reply by sanojsilva** [reply]
Also try adding the specific url as origin `origin: 'https://domain.com'`

**Comment by adrianjguerrero** [comment]
did you fix it? i'm with the same issue
